packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/utils/forms.js

```diff
@@ -235,6 +235,12 @@ const forms = {
               ),
             enumName: yup.string().nullable(),
           });
+        case 'text':
+          return yup.object().shape({
+            ...commonShape,
+            ...fieldsThatSupportMaxAndMinLengthShape,
+            regex: yup.string().nullable(),
```

Comment:
You need to add a test to check the regex value is valid. check that you can create a new Regexp without it throwing should be enough